File: AndroidXPatternsApp/jmvvm-app/src/main/java/com/woowrale/jmvvm/di/modules/MainModule.java
package com.woowrale.jmvvm.di.modules;

import android.content.Context;

import com.woowrale.jmvvm.MainApplication;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class MainModule {

    private final MainApplication mainApplication;

    public MainModule(MainApplication mainApplication) {
        this.mainApplication = mainApplication;
    }

    @Provides
    @Singleton
    Context provideContext(){
        return mainApplication;
    }

}

File: AndroidXPatternsApp/kmvvm-app/src/main/java/com/woowrale/kmvvm/di/modules/MainModule.kt
package com.woowrale.kmvvm.di.modules

import android.content.Context
import com.woowrale.kmvvm.MainApplication
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class MainModule(val mainApplication: MainApplication) {

    init {
        val mainApplication: MainApplication
    }

    @Provides
    @Singleton
    fun provideContext(): Context {
        return mainApplication
    }
}
